unison-src/transcripts/namespace-directive.md

```diff
@@ -62,14 +62,17 @@ type Baz = { qux : Nat }
 type RefersToFoo = RefersToFoo Foo
 
 refersToBar = cases
-  Bar -> 17
+  Foo.Bar -> 17
```

Comment:
This was ambiguous because it could have been [ns.]longer.foo.Foo.Bar or [file.]Foo.Bar, TDNR doesn't help, and so you have to expand it until it's unambiguous or an exact match?

Reply:
Correct. TDNR doesn't apply here – it's not used for patterns. 

Spelling out `Foo.Bar` would even be necessary if there was an explicit type signature on `refersToBar`.

In principle the type checker could of course help in that case; only a unique suffix among all the constructors of the known type should be required. But we don't make it to the type checker because we eagerly resolve all constructor references in patterns during parsing.